src/core/SampleClip.cpp

```diff
@@ -104,7 +90,20 @@ SampleClip::~SampleClip()
 	}
 }
 
+void SampleClip::setTrack(Track* track)
+{
+	Clip::setTrack(track);
 
+	if (getTrack())
+	{
+		disconnect(getTrack()->getMutedModel(), &Track::dataChanged, this, &SampleClip::playbackPositionChanged);
+	}
```

Comment:
Should this be done *before* you call `Clip::setTrack(track);`?

Reply:
`Clip::setTrack` is needed before `getTrack` is called I think.. However, the design still allows for certain kinds of misuse that I don't like, such setting the track of a clip than a different track it was added to. I want to prevent this from happening altogether.

I'll try to come back to this PR when I have time. There are also merge conflicts (most likely because of the merge of #7477), so I might want to restart.

Reply:
I'm not sure what the old code was doing, but it seems the point of this code is to disconnect the connection from the old track's muted model since you're not using that track anymore. So if you call `setTrack` first, it would disconnect the new track instead of the old track (and then connect it again), which doesn't make sense to me

Reply:
Oh yeah, you're totally right, have no idea what I was thinking. I think there's a better way to do this that isn't so error prone, so I'm going to try out some new ideas.